### ISSUE TITLE
Update users Email in Parse

### ISSUE THREAD

**Issue by arasu01** [ParsePlatform/Parse-SDK-iOS-OSX#51]
I have PFUser object and it has firstName, lastName and Email.

When I try to update the user's email which is already exist, i am receiving error. But when fetch the email from [PFUser currentUser], it returns wrong email, how to overcome this. Below is the my code for it.

PFUser *user = [PFUser currentUser];
user[@"email"] = @"test@gmail.com"
[user saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
       //UI refresh
    }];
I received the following error : the email address test@gmail.com has already been taken (Code: 203, Version: 1.7.4)

After this when fetching the email for current user

PFUser *user = [PFUser currentUser];
NSString *email = user[@"email"];
NSLog("email %@", email);
I received the following email "test@gmail.com" but which is wrong, which should be "tester@gmail.com".

FYI : I am also using parse offline store
This is posted in stack overflow in the below link : http://stackoverflow.com/questions/32064941/update-users-email-in-parse?answertab=active#tab-top

The same issue has been reported in the Parse Developers google forums : https://groups.google.com/forum/#!searchin/parse-developers/update$20email$20ios/parse-developers/CyH2wiywE7s/09KPXeXn7JwJ 

Waiting for a quick fix or workaround from Parse

**Comment by nlutsenko** [comment]
@arasu01 The error that you are seeing definetly comes from server.
The email field should be unique across the entire column, so if you already have a user with that email - you can't create or change another one to have the same one.

Hope it helps, if not - please reopen this issue or simply comment here.

**Comment by arasu01** [comment]
I was already aware that the error is coming from parse server.
In current user i have the email as "tester@gmail.com" and i was updating it the email to "test@gmail.com". And it gives the error from the parse server.

My issue arise from the current user where it should have the email as "tester@gmail.com", but whenever i fetch it is returning the "test@gmail.com", I think this is problem due to parse offline data store.

Here i have logout the current user and i have to log in again to get the correct email address.

This works fine when update the with the different address which didn't exists in Parse. Also when i am not using parse offline store.

**Comment by nlutsenko** [comment]
Thanks for clarifying, now it's clear.
The `email` field becomes dirty whenever you are changing it, and with Local Datastore - it will persist those changes locally, but indeed it will prevent from saving the user further, since that email already exists.

We don't have the `revert` functionality on a `PFObject` yet, but it's definitely on our list for things to be implemented. Let me create a separate issue for that one so we can mark it accordingly.

**Comment by nlutsenko** [comment]
@arasu01 Just created #52 for tracking this, feel free to comment there.

**Comment by arasu01** [comment]
@nlutsenko: Thanks for quick turn around.

**Comment by magneticorb** [comment]
I also came across this issue today. A PFUser had been updated with an email that belonged to another user, and couldn't save (obviously). After this point, calling fetchIfNeeded (even after unpinning) returned a dirty object. Also, logging the user out and logging them back in again returned a dirty object.

It seems the only way to get around this issue is to delete the app and reinstall it, which really isn't ideal - could I have gone about this any other way?
